healthcare/healthcare/doctype/patient_appointment/patient_appointment.py

```
@@ -631,6 +646,75 @@ def get_appointment_item(appointment_doc, item):
+	if patient_name and mobile:
+		match_conditions.append("(p.patient_name = %(patient_name)s and ifnull(p.mobile, '') = %(mobile)s)")
+
+	has_submitted_invoice = frappe.db.sql(
+		f"""
+		SELECT si.name
+		FROM `tabSales Invoice` si
+		LEFT JOIN `tabPatient` p ON p.name = si.patient
+		WHERE si.docstatus = 1
+			AND ({' OR '.join(match_conditions)})
+		LIMIT 1
+		""",
```

Ruff (0.15.4)

[error] Possible SQL injection vector through string-based query construction

(S608)